### PR TITLE
net: remove psock reference from connect

### PR DESCRIPTION
**Pull request by anchao**
## Summary

net: remove psock reference from connect

Signed-off-by: chao.an <anchao@xiaomi.com>

## Impact

N/A

## Testing

ci check

### CODE REVIEW

File: net/rpmsg/rpmsg_sockif.c [hunk 1]
```diff
@@ -287,11 +286,11 @@ static int rpmsg_socket_ept_cb(FAR struct rpmsg_endpoint *ept,
...
-      if (conn->psock)
-        {
-          conn->sconn.s_flags |= _SF_CONNECTED;
-          _SO_SETERRNO(conn->psock, OK);
-        }
+      conn->sconn.s_flags |= _SF_CONNECTED;
+#ifdef CONFIG_NET_SOCKOPTS
+      conn->sconn.s_error = OK;
+#endif
+      set_errno(OK);
```
Comment (xiaoxiang781216) [inline net/rpmsg/rpmsg_sockif.c:293]: why call set_errno here

Comment (anchao) [reply net/rpmsg/rpmsg_sockif.c:293]: Replace _SO_SETERRNO for equivalent, _SO_SETERRNO change the errno

Comment (xiaoxiang781216) [reply net/rpmsg/rpmsg_sockif.c:293]: should we create a new patch which change the first parameter of _SO_SETERRNO  to conn?

File: net/socket/net_sendfile.c [hunk 2]
```diff
@@ -133,8 +133,7 @@ ssize_t psock_sendfile(FAR struct socket *psock, FAR struct file *infile,
 
   if (ret < 0)
     {
-      FAR struct socket_conn_s *conn = psock->s_conn;
-      conn->s_error = -ret;
+      _SO_CONN_SETERRNO((FAR struct socket_conn_s *)psock->s_conn, -ret);
```
Comment (xiaoxiang781216) [inline net/socket/net_sendfile.c:136]: move the cast to _SO_CONN_SETERRNO

File: net/tcp/tcp_monitor.c [hunk 3]
```diff
@@ -165,10 +165,7 @@ static uint16_t tcp_monitor_event(FAR struct net_driver_s *dev,
 
           /* Clear the socket error */
 
-#ifdef CONFIG_NET_SOCKOPTS
-          conn->sconn.s_error = OK;
-#endif
-          set_errno(OK);
+          _SO_CONN_SETERRNO(&conn->sconn, OK);
```
Comment (xiaoxiang781216) [inline net/tcp/tcp_monitor.c:168]: _SO_CONN_SETERRNO(conn, OK);

File: net/tcp/tcp_netpoll.c [hunk 4]
```diff
@@ -128,10 +128,7 @@ static uint16_t tcp_poll_eventhandler(FAR struct net_driver_s *dev,
               reason = ECONNREFUSED;
             }
 
-#ifdef CONFIG_NET_SOCKOPTS
-          info->conn->sconn.s_error = reason;
-#endif
-          set_errno(reason);
+          _SO_CONN_SETERRNO(&info->conn->sconn, reason);
```
Comment (xiaoxiang781216) [inline net/tcp/tcp_netpoll.c:131]: _SO_CONN_SETERRNO(info->conn, reason);

File: net/socket/socket.h [hunk 5]
```diff
@@ -94,18 +94,30 @@
...
+    while (0)
+
 #  define _SO_SETERRNO(s,e) \
     do \
       { \
         if (s != NULL && (s)->s_conn != NULL) \
           { \
             FAR struct socket_conn_s *_conn = (s)->s_conn; \
-            _conn->s_error = (int16_t)e; \
+            _SO_CONN_SETERRNO(_conn, e); \
```
Comment (xiaoxiang781216) [inline net/socket/socket.h:111]: _SO_CONN_SETERRNO((s)->s_conn, e);
and remove line 109

File: net/socket/socket.h [hunk 6]
```diff
@@ -94,18 +94,30 @@
...
+      { \
+        (c)->s_error = (int16_t)e; \
+        set_errno(e); \
+      } \
+    while (0)
+
 #  define _SO_SETERRNO(s,e) \
     do \
       { \
         if (s != NULL && (s)->s_conn != NULL) \
```
Comment (xiaoxiang781216) [inline net/socket/socket.h:108]: let's move (s)->s_conn check to _SO_CONN_SETERRNO

File: net/can/can_recvmsg.c [hunk 7]
```diff
@@ -608,7 +606,7 @@ ssize_t can_recvmsg(FAR struct socket *psock, FAR struct msghdr *msg,
     }
 #endif
 
-  state.pr_sock   = psock;
+  state.pr_conn   = conn;
```
Comment (xiaoxiang781216) [inline net/can/can_recvmsg.c:609]: remove extra space before =

File: net/rpmsg/rpmsg_sockif.c [hunk 8]
```diff
@@ -287,11 +286,9 @@ static int rpmsg_socket_ept_cb(FAR struct rpmsg_endpoint *ept,
...
       conn->sendsize = head->size;
 
-      if (conn->psock)
-        {
-          conn->sconn.s_flags |= _SF_CONNECTED;
-          _SO_SETERRNO(conn->psock, OK);
-        }
+      conn->sconn.s_flags |= _SF_CONNECTED;
+
+      _SO_CONN_SETERRNO(&conn->sconn, OK);
```
Comment (xiaoxiang781216) [inline net/rpmsg/rpmsg_sockif.c:291]: _SO_CONN_SETERRNO(conn, OK);

File: net/socket/socket.h (hunk 6 above)
Comment (anchao) [reply net/socket/socket.h:108]: Done

File: net/socket/socket.h (hunk 5 above)
Comment (anchao) [reply net/socket/socket.h:111]: Done

File: net/rpmsg/rpmsg_sockif.c (hunk 8 above)
Comment (anchao) [reply net/rpmsg/rpmsg_sockif.c:291]: Done

File: net/tcp/tcp_netpoll.c (hunk 4 above)
Comment (anchao) [reply net/tcp/tcp_netpoll.c:131]: Done

File: net/tcp/tcp_monitor.c (hunk 3 above)
Comment (anchao) [reply net/tcp/tcp_monitor.c:168]: Done

File: net/socket/net_sendfile.c (hunk 2 above)
Comment (anchao) [reply net/socket/net_sendfile.c:136]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**